### DISCUSSION TITLE
How can i add new module to Sidebar menu?

### DISCUSSION THREAD

**Discussion by Shaykhnazar** [nasirkhan/laravel-starter#270] [Q&A]
I have created new module and permissions for this modules and attached to admin permissions. But I can't on the sidebar new module! Or have other requirements?

**Comment by nasirkhan** [comment]
Backend sidebar menu items are added via Middleware. You can check the other modules as an example:  

https://github.com/nasirkhan/laravel-starter/blob/master/Modules/Tag/Http/Middleware/GenerateMenus.php

**Reply by ajaykumishra** [reply]
Add GenerateMenus.php in my module middleware file still, the menu is not shown. Please help

**Reply by ncdfilho** [reply]
In case you didn't find it: 
Make sure you have added you new Module Middleware into the Provider's boot.

> Modules -> <yourmodule> -> Providers -> <yourmodule>ServiceProvider.php

$kernel->pushMiddleware('Modules\<yourmodule>\Http\Middleware\GenerateMenus');

**Reply by tarangpatel61** [reply]
> In case you didn't find it: Make sure you have added you new Module Middleware into the Provider's boot.
> 
> > Modules ->  -> Providers -> ServiceProvider.php
> 
> $kernel->pushMiddleware('Modules<yourmodule>\Http\Middleware\GenerateMenus');

Hi i have same issue  and i check my service provider as well it is already added the line which you mention but still not working
